### DISCUSSION TITLE
How to use the load_only() query option with 2.0

### DISCUSSION THREAD

**Discussion by miriamrozenbaum** [sqlalchemy/sqlalchemy#10731] [Usage Questions] [answered]
Hi,

As a part of the migration from SQLAlchemy 1.4 to 2.0 I'm trying to find a way to covert the Row list returns when using `session.execute(query).all()` to single ORM objects the way Query does in 1.4 version
In the documentation I saw the suggestion of using scalars but this method only works as expected when preforming select on the table and not when preforming select on specific list of columns of the table

thanks

**Comment by gordthompson** [comment]
Can you provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of a 1.4 query that you want to implement in 2.0?

**Comment by CaselIT** [comment]
Hi,

> In the documentation I saw the suggestion of using scalars but this method only works as expected when preforming select on the table and not when preforming select on specific list of columns of the table

query did not return a orm objects when selecting specific columns, so I'm not sure how you want to get orm objects from it.

Regarding the other part of the question, if you can't call `scalars()` before calling `all()` you may do something like
```py
rows = session.execute(query).all()
objs = [row[0] for row in rows]
```

**Reply by miriamrozenbaum** [reply]
first, thank you for your replies 
I figured out the root cause of my problem from @CaselIT your answer
My query in 1.4 used load_only with a dynamic list of columns passed in as a parameter. something like this:

```
    def get_requests_by_owner(self, *,
                              owner: str,
                              only: Optional[list[str]] = None) -> list[Request]:

        query = self.db_session.query(Request).filter(Request.owner == owner)

        if only:
            query = query.options(load_only(*only))

        return query.all()

```

Migrate the code to 2.0 style should change the code to look like that:


```
      def get_requests_by_owner(self, *,
                          owner: str,
                          only: Optional[list[str]] = None) -> list[Request]:
      
        query = select(Request).filter(Request.owner == owner)

        if only:
            query = query.options(load_only(*only))

        return self.db_session.execute(query).all()

```
in this case I could use the scalars method to get the list of request objects as it used to be in 1.4 version, but, of course, new load_only is no longer supporting this usage form in the 2.0 version, with DeprecationWarning flag I'm getting this warning:
`RemovedIn20Warning: Using strings to indicate column or relationship paths in loader options is deprecated and will be removed in SQLAlchemy 2.0.  Please use the class-bound attribute directly. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
`
as a solution I tried having the select query narrowing the fields down but now I get that it actually gives totally different results and it's not what I'm looking for actually.

So, the actual question is how can I use the load_lonly or any other solution in this dynamic way in SQLAlchemy 2.0

Thanks

**Reply by CaselIT** [reply, marked as answer]
you can do `query = query.options(load_only(*(getattr(Request, attr) for attr in only)))`

**Reply by miriamrozenbaum** [reply]
thank you
